src/pages/movies/index.tsx

```diff
@@ -1,91 +1,77 @@
 import {NextPage, GetServerSideProps} from "next"
 import MovieList from "../../components/MovieList"
 import PreviousPageButton from "../../components/PreviousPageButton"
+import FilterSelect from "../../components/FilterSelect"
 import {Movie} from "../../types"
 import classes from "./index.module.css"
+import { useState } from "react"
+import {useRouter} from "next/router"
+import {CustomButton} from "../../components/CustomMUI/CustomUI"
 
 
-//Hämtar data via API,
 
-/* export const getServerSideProps: GetServerSideProps = async (context) => {
-    const res = await fetch(`http://localhost:3000/api/movies`)
+export const getServerSideProps: GetServerSideProps = async (context) => {
+    const sort = context.query.sort
+    const size = context.query.size
+
+    const filters = []
+    if (sort) {
+        filters.push(`sort=${sort}`)
+    }
+    if (size) { 
+        filters.push(`size=${size}`)
+    }
+    const protocol = context.req.headers["x-forwarded-proto"] || "http";
+    const baseUrl = context.req ? `${protocol}://${context.req.headers.host}` : "";
+    const res = await fetch(`${baseUrl}/api/movies?${filters.join('&')}`)
     const data = await res.json()
-    
+
     return {
         props: {
             movies: data
         }
     }
-  }  */
+  } 
 
-export const Data = [
-    {
-        id: "1",
-        title: "Poch1",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "2",
-        title: "Poch2",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "3",
-        title: "Poch3",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "4",
-        title: "Poch4",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "5",
-        title: "Poch5",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },{
-        id: "6",
-        title: "Poch6",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    }
-    ,{
-        id: "7",
-        title: "Poch7",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "8",
-        title: "Poch8",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },{
-        id: "9",
-        title: "Poch9",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    }
-    ,{
-        id: "10",
-        title: "Poch10",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
+const Movies: NextPage<{movies: Movie[]}> = ({movies}) => {
+    const [filter, setFilter] = useState({
+        sort: "",
+        size: 9
+    })
```

Comment:
This state is stored in two places – the URL and this state tuple. This means that you have derived state, that can get out of sync. If the URL should always reflect the filter state, it can be stored just in the URL and then retrieved from there using `useRouter().query` instead of storing it in component state with `useState()`.

That way you know that you always only have the state in one place (the URL) and other parts of the code will respect that state, and as such always be in sync.